### PR TITLE
Skip authentication on listing ideas

### PR DESCRIPTION
**Pull request by JamesPartaker**
I am not sure if I need a test for this, let me know.

### CODE REVIEW

**Comment by swalberg** [comment]
You should be able to delete this and have https://github.com/winnipegrb/winnivote/blob/master/spec/features/list_ideas_spec.rb#L5-L7 a proper test.

**Comment by JamesPartaker** [comment]
This commit is in reference to Issue #16 that says a user shouldn't be required to have an account to view the list of ideas. Perhaps I misunderstood what the purpose was? 

Also I realized that this breaks the sign_in_spec test which relies on redirecting from the root_path when a user is not logged in.

**Comment by swalberg** [comment]
Sorry, what I mean is that list_ideas_spec already that logs in and looks for the ideas on the home page. If you remove the part where it logs in, the test should still pass, and also validate that you can view the page without authentication.

For sign_in_spec, you'll want to update the tests to reflect the new behaviour. Instead of relying on (and testing for) the redirect, you'll want to `click_link 'foo'` in order to get to the right place.

The (controller spec)[https://github.com/winnipegrb/winnivote/blob/master/spec/controllers/ideas_controller_spec.rb#L7] also logs the user in, so this should only happen for the appropriate actions.

So in short, your code is correct, but the tests need to be updated to reflect the new behaviour you've introduced.

**Comment by JamesPartaker** [comment]
Thanks for the help! I'll get working on modifying the tests.

**Comment by swalberg** [comment]
Sorry @JamesPartaker I missed your update. Thanks for making the fixes. :shipit:

**Comment by swalberg** [comment]
p.s. It's not necessary to fix in this PR but have a quick look at http://betterspecs.org/#let and http://betterspecs.org/#expect for some ways to improve your rspec-fu.

**Comment by JamesPartaker** [comment]
Thanks. I'll admit I didn't have a great idea about how the tests were to be written. I think that there is another issue to improve the tests altogether so I left them to be improved by another pull request. I have since gone through the RSPEC course on code school which was really helpful.

**Comment by amirci** [comment]
I think the ideas controller spec needs to check what happens when the user is not signed in and calls the methods that require sign in. Something like:

```
context "When user is not signed in" do
   describe '#update' do
       # spec here to specify error, we could use a macro perhaps
   end
end
```

Also I would refactor the instance variables like "@ideas" or "@user" to use **let**, for example:

```
let(:user) { FactoryGirl.create .... }
before { sign_in user }
```

**Comment by swalberg** [comment]
@amirci totally agree on both points but that might be better left for another PR. In particular, with the way the tests are structured we'll need `let!` as opposed to `let`.

**Comment by amirci** [comment]
I don't mind leaving the test refactoring for another PR but I do really think the tests about the scenario when the user is not signed in should be there.

**Comment by swalberg** [comment]
Let's finish this one up tonight!
